plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -78,17 +78,39 @@ import { Config } from '@backstage/config';
 import { Logger } from 'winston';
 import { connectEntityProviders } from '../processing/connectEntityProviders';
 import { permissionRules as catalogPermissionRules } from '../permissions/rules';
-import { PermissionAuthorizer } from '@backstage/plugin-permission-common';
+import {
+  Permission,
+  PermissionAuthorizer,
+} from '@backstage/plugin-permission-common';
 import {
   PermissionRule,
   createConditionTransformer,
   createPermissionIntegrationRouter,
 } from '@backstage/plugin-permission-node';
 import { AuthorizedEntitiesCatalog } from './AuthorizedEntitiesCatalog';
 import { basicEntityFilter } from './request/basicEntityFilter';
-import { RESOURCE_TYPE_CATALOG_ENTITY } from '@backstage/plugin-catalog-common';
+import {
+  catalogEntityCreatePermission,
+  catalogEntityDeletePermission,
+  catalogEntityReadPermission,
+  catalogEntityRefreshPermission,
+  catalogLocationCreatePermission,
+  catalogLocationDeletePermission,
+  catalogLocationReadPermission,
+  RESOURCE_TYPE_CATALOG_ENTITY,
+} from '@backstage/plugin-catalog-common';
```

Comment:
Perhaps we should consider looking for some neat export pattern that helps import all the permissions at once here? Perhaps something like always exporting an object record of permissions along with each one individually? That way we can use `Object.values(permissions)` here but people can also make use of it to neatly work with multiple permissions at once via e.g. `permissions.catalogEntityCreatePermission`.

Reply:
I actually thought the same thing when writing the PR originally, but doing so introduces a bunch of changes in a ton of files that are unrelated to the focus of this PoC, so I left it out. Definitely would do something like this for the actual implementation though!